spark-common/src/main/js/gulpfile.js

```diff
@@ -1,49 +1,10 @@
 var gulp = require('gulp');
 var karma = require('karma');
 var gutil = require('gulp-util');
-var soynode = require('gulp-soynode');
-var css2js = require('gulp-css2js');
-var concat = require('gulp-concat');
 var del = require('del');
-var replace = require('gulp-replace');
+var webpack = require('webpack');
 
-var distDir = 'target/dist';
-if (process.env.DIST_DIR) {
-    distDir = process.env.DIST_DIR;
-}
-
-var version = 'dev-version';
-if (process.env.PROJ_VERSION) {
-    version = process.env.PROJ_VERSION;
-}
-
-gulp.task('prepare-soy', function() {
-    // add the autoescape mode to the soy namespace that will produce an output matching the one
-    // from the soy compiler Atlassian uses
-    // the source file has to stay without the autoescape attribute to avoid problems with add-ons
-    // using the soy template directly as a webresource
-    return gulp.src('src/**/*.soy')
-        .pipe(replace(/\{namespace ([^\}]*)\}/, '{namespace $1 autoescape="deprecated-contextual"}'))
-        .pipe(gulp.dest('target/gen'));
-
-});
-
-gulp.task('compile-soy', ['prepare-soy'], function() {
```

Comment:
Removing compile-soy task breaks the 'test' and 'tdd' tasks
It should be possible to run the JS-code unit tests by running eg. 'yarn gulp test' in the JS code folder
At least there should not be broken tasks in the gulpfile